org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/wizards/NewJavaProjectWizardPageOne.java

```diff
@@ -1440,6 +1579,31 @@ public boolean getCreateModuleInfoFile() {
 		return fModuleGroup.getCreateModuleInfoFile();
 	}
 
+	/**
+	 * Returns the module name given.
+	 *
+	 * @return 'Module name:' text field value, removed of all spaces. If empty , returns
+	 * the project name, removed of all spaces.
+	 * @since 3.27
+	 */
+	public String getModuleName() {
+		String moduleName = fModuleGroup.getModuleName();
+		if (moduleName == null || moduleName.length() == 0) {
+			moduleName= fNameGroup.getName().replaceAll(WHITE_SPACE_STR, EMPTY_STR);
```

Comment:
Do we need to do this or just throw invalid error like we do in Create module-info dialog?

Reply:
Looks good now.